dotcom-rendering/src/web/components/FetchCommentCounts.importable.tsx

```diff
@@ -0,0 +1,173 @@
+import { useState } from 'react';
+import { renderToString } from 'react-dom/server';
+import { formatCount } from '../lib/formatCount';
+import { useApi } from '../lib/useApi';
+import { CardCommentCount } from './CardCommentCount';
+
+type Props = {
+	repeat?: boolean;
+};
+
+type RawCountType = {
+	id: string;
+	count: number;
+};
+
+type EnhancedCountType = {
+	id: string;
+	long: string;
+	short: string;
+	format: ArticleFormat;
+};
+
+type RenderedCountType = {
+	id: string;
+	html: string;
+	short: string;
+};
+
+/**
+ * @description
+ * Reads the dom looking for markers that contain the information needed to render the
+ * count
+ */
+function extractMarkers() {
+	const markers: { discussionId: string; format: ArticleFormat }[] = [];
+	document
+		.querySelectorAll('[data-name="comment-count-marker"]')
+		.forEach((element: Element) => {
+			if (element instanceof HTMLElement) {
+				const { discussionid: discussionId, format } = element.dataset;
+				if (discussionId && format)
+					markers.push({ discussionId, format: JSON.parse(format) });
+			}
+		});
+	return markers;
+}
+
+/**
+ * @description
+ * Creates the url used to fetch the count data
+ */
+function buildUrl() {
+	const markers = extractMarkers();
+	const discussionIds = markers.flatMap((marker) => marker.discussionId); // E.g.: ['/p/3pm9v', '/p/4k83z', '/p/6bnba']
+	return `https://api.nextgen.guardianapps.co.uk/discussion/comment-counts.json?shortUrls=${discussionIds.join(
+		',',
+	)}`;
+}
+
+/**
+ * @description
+ * Converts the count number into a formatted string and parses the stringified
+ * format
+ *
+ * E.g.:
+ *   - The 'long' version of 12345 is 12,345
+ *   - The 'short' version of 12345 is 12k
+ */
+function enhanceCounts(counts: RawCountType[]): EnhancedCountType[] {
+	const markers = extractMarkers();
+	return counts.map((count) => {
+		const { long, short } = formatCount(count.count);
+		// We don't get format in the api response so look it up in the array
+		// of DOM markers
+		const format = markers.find(
+			(marker) => marker.discussionId === count.id,
+		)?.format as ArticleFormat; // We cast because we're sure we will find it
+		return {
+			id: count.id,
+			long,
+			short,
+			format,
+		};
+	});
```

Comment:
If we’re not using the count object itself, we can de-structure here:

```suggestion
	return counts.map(({ count, id }) => {
		const { long, short } = formatCount(count);
		// We don't get format in the api response so look it up in the array
		// of DOM markers
		const format = markers.find(
			(marker) => marker.discussionId === id,
		)?.format as ArticleFormat; // We cast because we're sure we will find it
		return {
			id,
			long,
			short,
			format,
		};
	});
```

Reply:
Much nicer!